src/xmipp/libraries/reconstruction/tomo_filter_coordinates.cpp

```diff
@@ -0,0 +1,232 @@
+/***************************************************************************
+ *
+ * Authors:    Federico P. de Isidro-Gómez    fp.deisidro@cnb.csic.es (2021)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#include "tomo_filter_coordinates.h"
+#include <core/metadata_extension.h>
+#include <core/metadata_vec.h>
+#include <iostream>
+#include <core/utils/time_utils.h>
+
+
+// --------------------------- INFO functions ----------------------------
+
+void ProgTomoFilterCoordinates::readParams()
+{
+	fnInTomo = getParam("--inTomo");
+	fnMask = getParam("--mask");
+	fnInCoord = getParam("--coordinates");
+	radius = getIntParam("--radius");
+   	fnOut = getParam("-o");
+}
+
+
+void ProgTomoFilterCoordinates::defineParams()
+{
+	addUsageLine("This program generates a new metatada coordinates file with local information given by the input tomogram (e.g. local ");
+	addUsageLine("resolution map). \n"); 
+	addUsageLine("Given a set of coordinates and a tomogram, local statistics will be calculated for each coordinate (given by the radius). "); 
+	addUsageLine("Optionally a mask can be provided, and then those coordinates outside of the mask (value = 0) will be removed.");
+	
+	addParamsLine("  --inTomo <mrcs_file=\"\">                     : Input volume (mask or resolution map).");
+	addParamsLine("  [--mask <xmd_file=\"\">]                      : Input xmd file containing the 3D coordinates.");
+	addParamsLine("  --coordinates <xmd_file=\"\">                 : Percentile resolution threshold.");
```

Comment:
mismatch between `--coordinates` and `percentile`

Reply:
Done